reactor-core/src/main/java/reactor/util/Metrics.java

```diff
@@ -48,30 +59,89 @@ public class Metrics {
 	 *
 	 * @return true if the Micrometer instrumentation facade is available
 	 */
+	@Deprecated
 	public static final boolean isInstrumentationAvailable() {
 		return isMicrometerAvailable;
 	}
 
+	@Deprecated
 	public static class MicrometerConfiguration {
 
-		private static MeterRegistry registry = globalRegistry;
+		private static final Logger LOGGER = Loggers.getLogger(MicrometerHttpClientInterceptor.class);
+
+		private static final Method USE_REGISTRY;
+		private static final Method GET_REGISTRY;
+		private static final Method METRICS;
+
+		static {
+			Method useRegistry = null;
+			Method getRegistry = null;
+			Method metrics = null;
+			if (Metrics.reactorMicrometerMetricsClass != null) {
+				try {
+					useRegistry = Metrics.reactorMicrometerMetricsClass.getDeclaredMethod("useRegistry", MeterRegistry.class);
+					getRegistry = Metrics.reactorMicrometerMetricsClass.getDeclaredMethod("getRegistry");
+					metrics = Metrics.reactorMicrometerMetricsClass.getDeclaredMethod("metrics");
+				}
+				catch (NoSuchMethodException e) {
+					LOGGER.warn("Couldn't load reactor-metrics-micrometer, is it on the classpath?", e);
+					useRegistry = null;
+					getRegistry = null;
+					//metrics is already null if we get there because the metrics = getDeclaredMethod call is the last one that can fail
+				}
+			}
+			USE_REGISTRY = useRegistry;
+			GET_REGISTRY = getRegistry;
+			METRICS = metrics;
+		}
 
 		/**
 		 * Set the registry to use in reactor for metrics related purposes.
 		 * @return the previously configured registry.
 		 */
+		@Deprecated
 		public static MeterRegistry useRegistry(MeterRegistry registry) {
-			MeterRegistry previous = MicrometerConfiguration.registry;
-			MicrometerConfiguration.registry = registry;
-			return previous;
+			try {
+				return (MeterRegistry) USE_REGISTRY.invoke(null, registry);
+			}
+			catch (Exception e) {
+				LOGGER.warn("Using deprecated Metrics.MicrometerConfiguration#useRegistry without reactor-metrics-micrometer on the classpath", e);
+				return globalRegistry;
+			}
 		}
 
 		/**
 		 * Get the registry used in reactor for metrics related purposes.
 		 * @see Flux#metrics()
 		 */
+		@Deprecated
 		public static MeterRegistry getRegistry() {
-			return MicrometerConfiguration.registry;
+			try {
+				return (MeterRegistry) GET_REGISTRY.invoke(null);
+			}
+			catch (Exception e) {
+				LOGGER.warn("Using deprecated Metrics.MicrometerConfiguration#getRegistry without reactor-metrics-micrometer on the classpath", e);
+				return globalRegistry;
+			}
+		}
+
+		@Deprecated
+		public static final <T> SequenceListenerFactory<T, ?> metrics() {
```

Comment:
that one is unfortunate: in order to `Flux#metrics()` to delegate to `listen()` + the module's `SequenceListener`, we have to introduce this immediately deprecated public method.
that's the reason why I would step back and consider all the `Metrics` and `Flux#metrics()` infrastructure as active if and only if micrometer is on the classpath BUT THE MODULE ISN'T